From cc29e312925235aeab6aba6afec5ff357e39155a Mon Sep 17 00:00:00 2001
Message-Id: <cc29e312925235aeab6aba6afec5ff357e39155a@dist-git>
From: Peter Krempa <pkrempa@redhat.com>
Date: Thu, 12 Nov 2015 07:45:26 +0100
Subject: [PATCH] qemu: migration: Properly parse memory hotplug migration flag

https://bugzilla.redhat.com/show_bug.cgi?id=1280419 [7.2.z]

Since the flag was not enabled when 'eating' the migration cookie,
libvirt reported a bogus error when memory hotplug was enabled:

 unsupported migration cookie feature memory-hotplug

The error was ignored though due to a bug in the code so it slipped
through testing.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1278404
(cherry picked from commit f59808b724f99d43d042600455f55428a88a1ad9)

Signed-off-by: Jiri Denemark <jdenemar@redhat.com>
---
 src/qemu/qemu_migration.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/qemu/qemu_migration.c b/src/qemu/qemu_migration.c
index d3e024e..8074156 100644
--- a/src/qemu/qemu_migration.c
+++ b/src/qemu/qemu_migration.c
@@ -3350,7 +3350,8 @@ qemuMigrationPrepareAny(virQEMUDriverPtr driver,
 
     if (!(mig = qemuMigrationEatCookie(driver, vm, cookiein, cookieinlen,
                                        QEMU_MIGRATION_COOKIE_LOCKSTATE |
-                                       QEMU_MIGRATION_COOKIE_NBD)))
+                                       QEMU_MIGRATION_COOKIE_NBD |
+                                       QEMU_MIGRATION_COOKIE_MEMORY_HOTPLUG)))
         goto cleanup;
 
     if (STREQ_NULLABLE(protocol, "rdma") &&
-- 
2.6.3

